keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+
+import tensorflow as tf
+
+from keras_cv import core
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer, )
+from keras_cv.utils import preprocessing
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RandomResizedCrop(BaseImageAugmentationLayer):
+    """Randomly crops a part of an image and resizes it to provided size.
+
+    This implementation takes an intuitive approach, where we crop the images to a
+    random height and width, and then resize them. To do this, we first sample a
+    random value for area using `area_factor` and a value for aspect ratio using
+    `aspect_ratio_factor`. Further we get the new height and width by
+    dividing and multiplying the old height and width by the random area
+    respectively. We then sample offsets for height and width and clip them such
+    that the cropped area does not exceed image boundaries. Finally we do the
+    actual cropping operation and resize the image to `target_size`.
+
+    Args:
+        target_size: A tuple of two integers used as the target size to ultimately crop
+            images to.
+        area_factor: (Optional) A tuple of two floats, a single float or
+            `keras_cv.FactorSampler`. The ratio of area of the cropped part to
+            that of original image is sampled using this factor. Represents the
+            lower and upper bounds for the area relative to the original image
+            of the cropped image before resizing it to `target_size`.
+            `target_size`.  Defaults to (0.08, 1.0).
+        aspect_ratio_factor: (Optional) A tuple of two floats, a single float or
```

Comment:
I also don't think we can take a default here because this will be surprising to users.  That being said, a user SHOULD use this, so I suggest we provide no default and require this argument.

Reply:
@LukeWood do you mean to say for `aspect_ratio_factor`?

Reply:
@LukeWood In the case of a single float or integer, what should be the max (or min) value for `aspect_ratio_factor`? Currently we are doing this:
```py
if isinstance(aspect_ratio_factor, tuple):
  min_aspect_ratio = min(aspect_ratio_factor)
  max_aspect_ratio = max(aspect_ratio_factor)
elif isinstance(aspect_ratio_factor, core.FactorSampler):
  pass
else:
  raise ValueError(
      "Expected `aspect_ratio` to be tuple or FactorSampler. Received "
      f"aspect_ratio_factor={aspect_ratio_factor}."
  )
```

If we want to add a case for single ints and floats, we'll need to decide on a predefined max or min.

Reply:
lets just drop integer and float support for aspect ratio factor and make them pass a tuple.

Reply:
Awesome, thanks for clarifying!